Tests/Unit Tests/Modules/HeadlessUniversalCheckoutTests.swift

```diff
@@ -0,0 +1,431 @@
+        let apiConfiguration = setupApiConfiguration(
+            paymentMethod: paymentMethod,
+            surchargeAmount: surchargeAmount
+        )
+        setupMockApiClients(apiConfiguration: apiConfiguration)
+
+        let orderedExpectations = expectationsForDelegates(paymentMethod: paymentMethod, 
+                                                           handling: .manual,
+                                                           surchargeAmount: 99)
+
+        try presentNativeUIManager(paymentMethod: paymentMethod, expecting: orderedExpectations)
+    }
+
+    func testNativeUIManager_presentAdyenGiroPay_withoutSurcharge_auto_abort() throws {
+        setupSettings(handling: .auto)
+
+        let paymentMethod = Mocks.PaymentMethods.adyenGiroPayRedirectPaymentMethod
+        paymentMethod.baseLogoImage = PrimerTheme.BaseImage(colored: UIImage(), light: nil, dark: nil)
+        let apiConfiguration = setupApiConfiguration(
+            paymentMethod: paymentMethod
+        )
+        setupMockApiClients(apiConfiguration: apiConfiguration)
+
+        let orderedExpectations = expectationsForDelegates(paymentMethod: paymentMethod,
+                                                           shouldAbort: true)
+
+        try presentNativeUIManager(paymentMethod: paymentMethod, expecting: orderedExpectations)
+    }
+
+    // MARK: RawDataManager Tests
+
+    func testRawDataManager_presentPaymentCard_withoutSurcharge_auto() throws {
+        setupSettings(handling: .auto)
+
+        let paymentMethod = Mocks.PaymentMethods.paymentCardPaymentMethod
+        paymentMethod.baseLogoImage = PrimerTheme.BaseImage(colored: UIImage(), light: nil, dark: nil)
+        let apiConfiguration = setupApiConfiguration(
+            paymentMethod: paymentMethod
+        )
+        setupMockApiClients(apiConfiguration: apiConfiguration)
+
+        let orderedExpectations = expectationsForDelegates(paymentMethod: paymentMethod)
+
+        try submitWithRawDataManager(paymentMethod: paymentMethod, expecting: orderedExpectations)
+    }
+
+    func testRawDataManager_presentPaymentCard_withoutSurcharge_manual() throws {
+        setupSettings(handling: .manual)
+
+        let paymentMethod = Mocks.PaymentMethods.paymentCardPaymentMethod
+        paymentMethod.baseLogoImage = PrimerTheme.BaseImage(colored: UIImage(), light: nil, dark: nil)
+        let apiConfiguration = setupApiConfiguration(
+            paymentMethod: paymentMethod
+        )
+        setupMockApiClients(apiConfiguration: apiConfiguration)
+
+        let orderedExpectations = expectationsForDelegates(paymentMethod: paymentMethod,
+                                                           handling: .manual)
+
+        try submitWithRawDataManager(paymentMethod: paymentMethod, expecting: orderedExpectations)
+    }
+
+    func testRawDataManager_presentAdyenGiroPay_withoutSurcharge_auto_abort() throws {
```

Comment:
I may be confused, but from the old HUC tests, it looks like `rawDataManager` tests were only run with `paymentCard`. Has something changed here, or am I getting my wires crossed?